bot/role_management/role_management.py

```diff
@@ -45,7 +45,9 @@ async def toggle_module(self, ctx: commands.Context, *, str_modules: str):
             str_modules (str): A string containing abbreviations of all the modules a user would like to toggle.
         """
         if ctx.channel.id != self.ch_role.id:
-            await ctx.message.delete()
+            if not self._db_connector.is_botonly(ctx.channel.id):
+                await ctx.message.delete()
```

Comment:
Cant you connect these two with and?

Reply:
Why? The first if condition is to check if the message has not been in the configured roles channel. If yes, the second if checks if the current channel is in bot-only mode. If not, the invocation should be deleted, if yes, the message has probably already been deleted and trying againd would result in an exception.

Reply:
I mean write 

```
if ctx.channel.id != self.ch_role.id and not self._db_connector.is_botonly(ctx.channel.id):
                await ctx.message.delete()
```

This is semantically equivalent but more concise and exact.

Reply:
No its not. Just look at the identation of the whole code in the command. I think you're misreading it.